munit/src/main/scala/munit/MUnitRunner.scala

```diff
@@ -5,18 +5,17 @@ import org.junit.runner.notification.RunNotifier
 import java.lang.reflect.Modifier
 import org.junit.runner.notification.Failure
 import scala.util.control.NonFatal
-import org.junit.AssumptionViolatedException
 import org.junit.runner.manipulation.Filterable
 import org.junit.runner.manipulation.Filter
 import org.junit.runner.Runner
 import scala.collection.mutable
 
-class MUnitRunner(cls: Class[_ <: Suite]) extends Runner with Filterable {
+class MUnitRunner(val cls: Class[_ <: Suite]) extends Runner with Filterable {
   require(
     hasEligibleConstructor(),
     s"Class '${cls.getCanonicalName()}' is missing a public empty argument constructor"
   )
-  lazy val suite = cls.newInstance()
+  val suite = cls.newInstance()
```

Comment:
speaking of `lazy`, Dotty complains because `suite` is referenced in type position (e.g. `suite.Test`) so it can't be a `lazy` val.

```
[error] -- Error: /Users/gabro/buildo/munit/munit/src/main/scala/munit/MUnitRunner.scala:127:12
[error] 127 |      test: suite.Test
[error]     |            ^^^^^
[error]     | (MUnitRunner.this.suite : MUnitRunner.this.cls.T) is not a legal path
[error]     | since it refers to nonfinal lazy value suite
```

Making it `final` doesn't help:

```
[error] -- Error: /Users/gabro/buildo/munit/munit/src/main/scala/munit/MUnitRunner.scala:127:12
[error] 127 |      test: suite.Test
[error]     |            ^^^^^
[error]     |(MUnitRunner.this.suite : MUnitRunner.this.cls.T) is not a legal path
[error]     |since its underlying type (MUnitRunner.this.suite : MUnitRunner.this.cls.T) is not a concrete type
```

It intuitively make sense to me, and removing the `lazy` was the easiest thing to make it compile.

Reply:
I would prefer to keep the `lazy` modifier and instead case to `Any` or `val suite: Suite { type TestValue = Any }` or something like that.